### PR TITLE
Tox

### PR DESCRIPTION
**Pull request by davidszotten**
initial stab at using tox

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,8 @@
-# See https://bugs.launchpad.net/pyflakes/+bug/1223150
```
Comment (davidszotten) [inline Makefile]: maybe drop makefile altogether?

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    pinned: path.py==5.1
+    pinned: pyrabbit==1.1.0
+    pinned: Werkzeug==0.9.6
+
+commands =
+    pip install --editable .[dev]
+    flake8 nameko test
+    pylint --rcfile=pylintrc nameko -E
+    coverage run --concurrency=eventlet --source nameko -m \
+        pytest test {posargs:--timeout=30}
```
Comment (mattbennett) [inline tox.ini]: what does ``posargs`` do?

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    coverage report --fail-under=100
+
+
+[testenv:docs]
+commands =
+    pip install --editable .[docs]
+    sphinx-build -n -b html -d {envtmpdir}/doctrees docs {envtmpdir}/html
```
Comment (mattbennett) [inline tox.ini]: ``envtmpdir`` is presumably a tox provided thing?

File: tox.ini (hunk 2 above)
Comment (davidszotten) [inline tox.ini]: injects positional args passed to tox

File: tox.ini (hunk 3 above)
Comment (davidszotten) [inline tox.ini]: yep: http://tox.readthedocs.org/en/latest/config.html?highlight=envdir#confval-envdir

**Comment by mattbennett** [comment]
:+1:

**Pull request merged**